libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/target/CompositeRequestHelper.java

```diff
@@ -26,78 +26,109 @@
  */
 package com.salesforce.androidsdk.mobilesync.target;
 
-import com.salesforce.androidsdk.mobilesync.manager.SyncManager;
-import com.salesforce.androidsdk.mobilesync.util.Constants;
-import com.salesforce.androidsdk.rest.CompositeResponse;
-import com.salesforce.androidsdk.rest.CompositeResponse.CompositeSubResponse;
-import com.salesforce.androidsdk.rest.RestRequest;
-import com.salesforce.androidsdk.rest.RestResponse;
-import com.salesforce.androidsdk.util.JSONObjectHelper;
-
-import org.json.JSONException;
-import org.json.JSONObject;
-
-import java.io.IOException;
-import java.net.HttpURLConnection;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.Map;
+ import com.salesforce.androidsdk.mobilesync.manager.SyncManager;
+ import com.salesforce.androidsdk.mobilesync.util.Constants;
+ import com.salesforce.androidsdk.rest.CollectionResponse;
+ import com.salesforce.androidsdk.rest.CollectionResponse.CollectionSubResponse;
+ import com.salesforce.androidsdk.rest.CompositeResponse;
+ import com.salesforce.androidsdk.rest.CompositeResponse.CompositeSubResponse;
+ import com.salesforce.androidsdk.rest.RestRequest;
+ import com.salesforce.androidsdk.rest.RestResponse;
+ import com.salesforce.androidsdk.util.JSONObjectHelper;
+ import java.io.IOException;
+ import java.io.UnsupportedEncodingException;
+ import java.net.HttpURLConnection;
+ import java.util.HashMap;
+ import java.util.HashSet;
+ import java.util.LinkedHashMap;
+ import java.util.LinkedList;
+ import java.util.List;
+ import java.util.Map;
+ import org.json.JSONArray;
+ import org.json.JSONException;
+ import org.json.JSONObject;
 
 public class CompositeRequestHelper {
 
     /**
-     * Build and send composite request
+     * Send record requests using a composite batch request
      * @param syncManager
      * @param allOrNone
-     * @param refIdToRequests
-     * @return map of ref id to composite sub response
+     * @param recordRequests
+     * @return map of reference id to record responses
      * @throws JSONException
      * @throws IOException
      */
-     public static Map<String, CompositeSubResponse> sendCompositeRequest(SyncManager syncManager, boolean allOrNone, LinkedHashMap<String, RestRequest> refIdToRequests) throws JSONException, IOException {
```

Comment:
True. 
This method should not really be public - it's a util method used internally by some of the targets. Hopefully nobody used it.

Reply:
Perhaps we should deprecate and move it to protected/package private in 11.  Not high priority obviously since it has been this way for years already.